### PR TITLE
feat: qr code scanner

### PR DESCRIPTION
**Pull request by adityapawar1**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
Adding a page for QR code scanning with [expo-camera](https://docs.expo.dev/versions/latest/sdk/camera/). After scanning a QR code, it takes you to the placeholder Tree Info Page where it shows you the "id" of the tree you just scanned.
Also added some boilerplate for [react-navigation](https://reactnavigation.org/docs/getting-started) to pass params like the tree id between pages.

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."
QR Code Scanner:
Screenshot 2024-11-06 at 6 36 24 PM

## How to review
[//]: # "Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?"
Use a QR Code generator (like [this](https://www.qr-code-generator.com/)) and add a tree id as the value. Scan the QR code and see if it gets the correct data from the QR code, and navigates you to the tree info page.

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
Make everything look pretty!

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'
https://docs.expo.dev/versions/latest/sdk/camera/
https://reactnavigation.org/docs/typescript/#type-checking-screens

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"
None

CC: @christophertorres1

### CODE REVIEW

**Comment by AlexWang05** [comment]
LGTM 👍

File: src/types.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+// Not sure where to keep this stuff
```
Comment (christophertorres1) [inline src/types.ts:1]: Remove comments except short and precise ones for functions

**Review by christophertorres1 — CHANGES_REQUESTED** [review]
Amazing work! 

Just some minor changes to maintain consistency in the project's file structure, and remove extra comments except for function descriptions. 

Thanks Adi!!

**Comment by adityapawar1** [comment]
should be ready for review @christophertorres1

**Review by christophertorres1 — APPROVED** [review]
